From 75041cbc9e6cf7b134980126a28a914f4fe26dc9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E5=BF=98=E6=80=80?= <guanwentao@uniontech.com>
Date: Tue, 19 Nov 2024 03:08:18 +0000
Subject: [PATCH 285/360] config: deepin_arm64_desktop_defconfig: Enable
 LIST_HARDENED

Linux kernel 6.6 introduced CONFIG_LIST_HARDENED, of which CONFIG_DEBUG_LIST is a superset [1]. Since CONFIG_DEBUG_LIST selects CONFIG_LIST_HARDENED, configs that already selected CONFIG_DEBUG_LIST were not prompted for CONFIG_LIST_HARDENED.
Since CONFIG_DEBUG_LIST has never been designed with performance in mind, non-debug kernels should prefer CONFIG_LIST_HARDENED. The main benefit is improved performance; the only downside is slightly worse error reporting (the WARNING is turned into an access fault in some cases). Hardening-wise there is no functional difference.
[1] https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=aebc7b0d8d91bbc69e976909963046bc48bca4fd

Link: https://gitlab.archlinux.org/archlinux/packaging/packages/linux/-/issues/26
---
 arch/arm64/configs/deepin_arm64_desktop_defconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm64/configs/deepin_arm64_desktop_defconfig b/arch/arm64/configs/deepin_arm64_desktop_defconfig
index d61ff55f5763..81b4c7d2c760 100644
--- a/arch/arm64/configs/deepin_arm64_desktop_defconfig
+++ b/arch/arm64/configs/deepin_arm64_desktop_defconfig
@@ -4063,6 +4063,7 @@ CONFIG_IMA_WRITE_POLICY=y
 # CONFIG_IMA_READ_POLICY is not set
 CONFIG_IMA_APPRAISE=y
 CONFIG_DEFAULT_SECURITY_APPARMOR=y
+CONFIG_LIST_HARDENED=y
 CONFIG_CRYPTO_USER=m
 # CONFIG_CRYPTO_MANAGER_DISABLE_TESTS is not set
 CONFIG_CRYPTO_PCRYPT=m
-- 
2.20.1

